### DISCUSSION TITLE
Adding a new computed column in the arrow table

### DISCUSSION THREAD

**Discussion by zacqed** [duckdb/duckdb#3122] [Q&A] [answered]
In the below snippet, the output of con.execute needs to be stored into a table and then accessed for being able to append the column into arrow table. Is there any simpler way that we can use directly?

```
import duckdb
import pyarrow as pa
import pandas as pd
import pyarrow.parquet as pq

lineitem = pq.read_table('lineitemsf1.snappy.parquet')
con = duckdb.connect()
con.execute("PRAGMA threads=4")
con.execute("PRAGMA verify_parallelism")

temp = con.execute("""SELECT l_extendedprice * l_discount AS revenue FROM lineitem;""").fetch_arrow_table()
lineitem = lineitem.append_column(pa.field("revenue",lineitem["l_discount"].type), temp["revenue"])
```

Basically if we can remove temp creation and temp["revenue"].

**Comment by Mytherin** [comment]
Unrelated to the question, but I notice that you are using `verify_parallelism`. This is a debugging flag that decreases the base unit of parallelism significantly to allow for parallel execution on smaller data sets (where this would not be efficient). In general this should not be used outside of debugging the engine since it will increase conflicts between threads. Is there a specific reason that you are using this flag?

**Reply by zacqed** [reply]
Thanks for your response. Please avoid the line, it was included for debugging purpose.

**Comment by Alex-Monahan** [comment, marked as answer]
While this is untested, I believe this will produce the same result, but in a simpler way! Let me know if this is helpful or if you are looking for something different.

Essentially, you can just read the entire parquet file's contents alongside the calculated column in the same SQL query. Then there is no need to read in the parquet file ahead of time, or to add in a column!

```python
import duckdb
import pyarrow as pa

con = duckdb.connect()
con.execute("PRAGMA threads=4")

lineitem = con.execute("""
    SELECT 
        l_extendedprice * l_discount AS revenue
        , lineitem.* 
    FROM 'lineitemsf1.snappy.parquet' lineitem;
""").fetch_arrow_table()
```

**Reply by zacqed** [reply]
Thanks, let me give this method a bit more thought and come back.
